service/url_service.py

```diff
@@ -1,4 +1,65 @@
+from models.pynamodb_model import UrlEntry
+import uuid
 # This is where your URL shortening logic will go
-def generate_short_url(url: str, short_id_length: int) -> str:
+def generate_short_url(url: str, custom_url: str = None, short_id_length: int = 6) -> str:
```

Comment:
I'd recommend longer than 6 as the length. Using just the first 6 characters might still lead to a high collision count or can be brute forced. Find a good balance between security, performance and reaching the goal of "shortening urls". I'd suggest around 10-15 or so characters.